src/lib/stack-client.js

```diff
@@ -121,6 +121,51 @@ const fetchJSON = function(method, path, body, options={}) {
   )
 }
 
+/**
+ * Test if an error is from an HTTP 404
+ *
+ * @function
+ * @private
+ * @param {Function} error - received from a fetch
+ * @returns {boolean}
+ */
+const is404 = function(error) {
+  return ['NotFoundException', 'NotFound', 'FetchError'].includes(error.name) &&
+     error.status && 
+     error.status === 404
+} 
+
+/**
+ * Memoize the result of a function which does an HTTP fetch
+ * 
+ * If a call throws an error because the 
+ * underlying HTTP request returned a 404
+ * then this function returns a default value
+ * 
+ * In the absence of any other error, the result is
+ * cached and reused in the next call to the function.
+ * 
+ *
+ * @function
+ * @param {Function} fn - the function to memoize. It will be
+ *                        called without any parameter
+ * @param {Object} defaultValue - returned in case of 404
+ */                  
+const withCache = function(fn, defaultValue) {
+  let cache = undefined;
+  return function() {
+    if (cache === undefined) {
+      return fn().then(result => 
+        cache = result
+      ).catch( error => 
+        cache = is404(error) ? defaultValue : undefined
+      ) 
+    } else {
+      return Promise.resolve(cache)
+    }
+  }
+}
```

Comment:
IMHO the cache should be limited to the cozy you are on. It can be achievied by having the cache as an object and using the stackClient.uri as the key to the cache.

Reply:
lodash memoize comes to mind.

```
const key = () => getStackClient().uri
const context = memoize(x => undefinedIf404(getContext()), key)
```

Reply:
Is there a risk to have an execution going through multiple instance ? I don't see how

Reply:
On mobile, you can switch cozies

Reply:
It does not unload and reload the whole code ? I am reluctant to this because it changes the behaviour of the legacy client

Reply:
No in Banks, we do not reload the page after the logout.

Reply:
ok -> #471